docs/network-binding-plugin.md

```diff
@@ -0,0 +1,342 @@
+The following plugins are available:
+
+- [passt](https://kubevirt.io/user-guide/virtual_machines/net_binding_plugins/passt/) [v1.1.0]
+- [macvtap](https://kubevirt.io/user-guide/virtual_machines/net_binding_plugins/macvtap/) [v1.1.1]
+- [slirp](https://kubevirt.io/user-guide/virtual_machines/net_binding_plugins/slirp/) [v1.1.0]
+
+## The Zero Code Plugin
+
+The simplest network binding plugin is the one that requires no additional code.
+It takes advantage of the
+[Domain Attachment Type](https://kubevirt.io/user-guide/virtual_machines/network_binding_plugins/#domainattachmenttype)
+option which provides a pre-defined core Kubevirt method to attach an interface
+to the domain.
+
+The currently supported domain attachment type is `tap` (v1.1.1) which
+builds a domain interface configuration that points to the tap/macvtap
+existing interface.
+
+Such a binding plugin assumes that the CNI used for the network connectivity
+exposes in the pod a `tap` or `macvtap` (type) interface with a name corresponding
+to the hashed network name:
+
+- `pod<hash network name>` (or plain `eth0` for the primary network)
+- `tap<hash network name>` (or plain `tap0` for the primary network)
+
+Kubevirt virt-controller will define these names on the pod multus annotation,
+therefore there is no special action needed from the plugin author except
+choosing a CNI that confirms with the CNI standard spec.
+
+For clarification, the CNI used for the connectivity is the one defined
+in the `NetworkAttachmentDefinition` object referenced from the VM network spec.
+
+### Configuration example
+
+Given the following registered binding plugin:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    network:
+      Binding:
+        my-zero-code-binding:
+          domainAttachment: tap
+```
+
+The VM interface/network spec will include a ref to the binding plugin
+and the `NetworkAttachmentDefinition` named `mynad`:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: myvm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: mynetwork
+              binding:
+                name: my-zero-code-binding
+      networks:
+      - name: mynetwork
+        multus:
+          networkName: mmynad
+```
+
+The `mynad` NetworkAttachmentDefinition defines `cni-name` as the CNI plugin
+binary:
+
+```yaml
+kind: NetworkAttachmentDefinition
+apiVersion: k8s.cni.cncf.io/v1
+metadata:
+  name: mynad
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "name": "some-name",
+      "type": "cni-name",
+    }'
+```
+
+Kubevirt will eventually create a domain interface configuration that
+points to the pod interface (e.g. `pod12345678`).
+This is a snippet of the configuration:
+
+```xml
+<interface type='ethernet'>
+   <alias name='ua-mynetwork'/>
+   <target dev='pod12345678' managed='no'/>
```

Comment:
Please mention the target will be the tap name or the pod interface name if the tap doesn't exist.